Turing self evaluation: computing pillar (#228)
sa-tre/satre-specification: docs/source/evaluations/alan_turing_institute.md

:name: tab-turing-dsh-computing-code-vcs

* -
  - Score
  - Response
* - 2.1.11.
  - 1
  - - Version control tools are provided to users, including an internal GitLab instance.
    - Users are encouraged to version control their code and we provide training for those who are unfamiliar with git.
    ##### Potential Improvements
    - We do not provide specific tools to aid or encourage reproducibility or creating data analysis pipelines.
    - We do not support CI pipelines on our GitLab server.
    - We do not have a method to ensure that work done inside the environment can be reproduced outside such as containerisation.
* - **Capability met?**
  - **YES**
  -
```

## Artefact Management Application

```{list-table}
:header-rows: 1
:name: tab-turing-dsh-computing-artefact-management

* -
  - Score
  - Response
* - 2.1.12.
  - 2
  - We provide proxied access to external software repositories, currently PyPI and CRAN, using Sonatype Nexus.
    For our highest sensitivity projects we instead provide a local mirror.
    In either case, we can support either access to every package in the remote repository or a pre-specified allowed list of approved packages.
* - 2.1.13.
  - 2

>   - - We do restrict access to a pre-specified allowed list for higher sensitivity environments.
>       These allowed lists include a minimal set of well-used and useful packages plus their dependencies.
>     - We have guidelines about how to request new packages which involve making a request on our GitHub repository.
>       Requests are then approved or denied by sysadmins based on whether their usefulness outweighs any potential risk to the project.

Author: ```suggestion
  - - For higher sensitivity environments, we restrict access to a pre-specified allowed list.
      These allowed lists are configurable on a per-project basis and, by default, include a minimal set of well-used and useful packages plus their dependencies.
```

>       Requests are then approved or denied by sysadmins based on whether their usefulness outweighs any potential risk to the project.

Reviewer: Are we mixing up the process for core packages and package proxies here?

It might be clearer just to explain that the allowlists are configurable per project and we provide sensible defaults.

Author: Done

  - - We do have separate development environments, but these are not permanent clones of our production environment.
    - We do not automate promotion of development environments to production.
    ##### Potential Improvements
    - We would consider moving to a blue/green deployment environment as long as this is possible from a cost perspective.
* - **Capability met?**
  - **YES**
  -
```

## Infrastructure Removal Process

```{list-table}
:header-rows: 1
:name: tab-turing-dsh-computing-infrastructure-removal

* -
  - Score
  - Response
* - 2.2.6.
  - 2
  - We use Powershell scripts to automate the removal of unused infrastructure.
    We have documented procedures that detail when this should be done.
* - **Capability met?**
  - **YES**
  -
```

## Availability Management Process

```{list-table}
:header-rows: 1
:name: tab-turing-dsh-computing-infrastructure-availability-management

* -
  - Score
  - Response
* - 2.2.7.
  - 2
  - Azure publishes availability and uptime guarantees for relevant services.
    We have chosen replication levels which balance high availability while keeping data within a single region.
* - 2.2.8.
  - 0
  - We do not have an availability target.

>     We do not make any guarantees to our users.

Reviewer: ```suggestion
    We do not make any availability guarantees to our users.
```
